### PR TITLE
Added 'availability_template' to Template Binary Sensor platform

### PR DESCRIPTION
**Pull request by grillp**
Added documentation `availability_template` configuration option for Template Binary Sensor platform components.

https://github.com/home-assistant/home-assistant/pull/26510

## Checklist:

- [X] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [X] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

<a href="https://gitpod.io/#https://github.com/home-assistant/home-assistant.io/pull/10123"></a>

### CODE REVIEW

File: source/_components/binary_sensor.template.markdown [hunk 1]
```diff
@@ -58,6 +64,11 @@ sensors:
           description: The sensor is `on` if the template evaluates as `True` and `off` otherwise. The actual appearance in the frontend (`Open`/`Closed`, `Detected`/`Clear` etc) depends on the sensor’s device_class value
           required: true
           type: template
+        availability_template:
+          description: "Defines a template to get the `available` state of the component. If the template returns `true` the device is `available`. If the template returns any other value, the device will be `unavailable`. If `availability_template` is not configured, the component will always be `available`"
```
Comment (arsaboo) [inline source/_components/binary_sensor.template.markdown]: Remove the quotes around the entire description.

File: source/_components/binary_sensor.template.markdown [hunk 2]
```diff
@@ -58,6 +64,11 @@ sensors:
...
+          required: false
+          type: template
+          default: the device is always `available`
```
Comment (arsaboo) [inline source/_components/binary_sensor.template.markdown]: Should be `true`

**Review by arsaboo — DISMISSED** [review]

File: source/_components/binary_sensor.template.markdown (hunk 2 above)
Comment (grillp) [reply source/_components/binary_sensor.template.markdown]: As in:
`default: "true"`
or
`default: true`

File: source/_components/binary_sensor.template.markdown (hunk 1 above)
Comment (grillp) [reply source/_components/binary_sensor.template.markdown]: Will do. I'll have to update this in all 8 PRs.

File: source/_components/binary_sensor.template.markdown (hunk 2 above)
Comment (arsaboo) [reply source/_components/binary_sensor.template.markdown]: `default: true`

File: source/_components/binary_sensor.template.markdown [hunk 3]
```diff
@@ -27,7 +28,12 @@ binary_sensor:
         friendly_name: "Sun is up"
         value_template: >-
           {{ state_attr('sun.sun', 'elevation')|float > 0 }}
+        availability_template: >-
```
Comment (frenck) [inline source/_components/binary_sensor.template.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

File: source/_components/binary_sensor.template.markdown [hunk 4]
```diff
@@ -66,15 +77,6 @@ sensors:
           description: Defines a template for the entity picture of the sensor.
           required: false
           type: template
-        attribute_templates:
```
Comment (frenck) [inline source/_components/binary_sensor.template.markdown]: Why has this been removed?

File: source/_components/binary_sensor.template.markdown [hunk 5]
```diff
@@ -217,45 +228,17 @@ binary_sensor:
              or is_state('binary_sensor.porch_ms6_1_129', 'on')
              or is_state('binary_sensor.family_room_144', 'on') }}
 ```
-{% endraw %}
-
-### Device Tracker sensor with Latitude and Longitude Attributes
```
Comment (frenck) [inline source/_components/binary_sensor.template.markdown:222]: Why is this being removed? Seems unrelated.

File: source/_components/binary_sensor.template.markdown [hunk 6]
```diff
@@ -58,6 +64,11 @@ sensors:
           description: The sensor is `on` if the template evaluates as `True` and `off` otherwise. The actual appearance in the frontend (`Open`/`Closed`, `Detected`/`Clear` etc) depends on the sensor’s device_class value
           required: true
           type: template
+        availability_template:
+          description: Defines a template to get the `available` state of the component. If the template returns `true` the device is `available`. If the template returns any other value, the device will be `unavailable`. If `availability_template` is not configured, the component will always be `available`
```
Comment (frenck) [inline source/_components/binary_sensor.template.markdown]:
```suggestion
          description: Defines a template to get the `available` state of the component. If the template returns `true`, the device is `available`. If the template returns any other value, the device will be `unavailable`. If `availability_template` is not configured, the component will always be `available`.
```

**Review by frenck — CHANGES_REQUESTED** [review]
I've reviewed your PR, @grillp. Left some comments (a couple more this time). Could you take a look? Thanks! 👍

File: source/_components/binary_sensor.template.markdown (hunk 4 above)
Comment (grillp) [reply source/_components/binary_sensor.template.markdown]: Yeah, the attribute_template option was added to the code after I had submitted my original (joint) pull request 4 weeks ago. I'll add these lines back in.

File: source/_components/binary_sensor.template.markdown (hunk 5 above)
Comment (grillp) [reply source/_components/binary_sensor.template.markdown:222]: Sme as above. The 'attribute_attribute' feature was added after my original pull request 4 weeks ago.

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.
